fe/src/main/java/org/apache/doris/load/loadv2/dpp/GlobalDictBuilder.java

```diff
@@ -0,0 +1,413 @@
+    private SparkSession spark;
+
+    // key=doris column name,value=column type
+    private Map<String, String> dorisColumnNameTypeMap = new HashMap<>();
+
+    // column in this list means need split distinct value and then encode respectively
+    // to avoid the performance bottleneck to transfer origin value to dict value
+    private List<String> veryHighCardinalityColumn;
+    // determine the split num of new distinct value,better can be divisible by 1
+    private int veryHighCardinalityColumnSplitNum;
+
+    private ExecutorService pool;
+
+    private StructType distinctValueSchema;
+
+    public GlobalDictBuilder(MultiValueMap dictColumn,
+                             List<String> dorisOlapTableColumnList,
+                             List<String> mapSideJoinColumns,
+                             String sourceHiveDBTableName,
+                             String sourceHiveFilter,
+                             String dorisHiveDB,
+                             String distinctKeyTableName,
+                             String globalDictTableName,
+                             String dorisIntermediateHiveTable,
+                             int buildConcurrency,
+                             List<String> veryHighCardinalityColumn,
+                             int veryHighCardinalityColumnSplitNum,
+                             SparkSession spark) {
+        this.dictColumn = dictColumn;
+        this.dorisOlapTableColumnList = dorisOlapTableColumnList;
+        this.mapSideJoinColumns = mapSideJoinColumns;
+        this.sourceHiveDBTableName = sourceHiveDBTableName;
+        this.sourceHiveFilter = sourceHiveFilter;
+        this.distinctKeyTableName = distinctKeyTableName;
+        this.globalDictTableName = globalDictTableName;
+        this.dorisIntermediateHiveTable = dorisIntermediateHiveTable;
+        this.spark = spark;
+        this.pool = Executors.newFixedThreadPool(buildConcurrency < 0 ? 1 : buildConcurrency);
+        this.veryHighCardinalityColumn = veryHighCardinalityColumn;
+        this.veryHighCardinalityColumnSplitNum = veryHighCardinalityColumnSplitNum;
+
+        spark.sql("use " + dorisHiveDB);
+    }
+
+    public void createHiveIntermediateTable() throws AnalysisException {
+        Map<String, String> sourceHiveTableColumn = spark.catalog()
+                .listColumns(sourceHiveDBTableName)
+                .collectAsList()
+                .stream().collect(Collectors.toMap(Column::name, Column::dataType));
+
+        Map<String, String> sourceHiveTableColumnInLowercase = new HashMap<>();
+        for (Map.Entry<String, String> entry : sourceHiveTableColumn.entrySet()) {
+            sourceHiveTableColumnInLowercase.put(entry.getKey().toLowerCase(), entry.getValue().toLowerCase());
+        }
+
+        // check and get doris column type in hive
+        dorisOlapTableColumnList.stream().forEach(columnName -> {
+            String columnType = sourceHiveTableColumnInLowercase.get(columnName);
```

Comment:
if column name of doris table is upper, this will be wrong.
Maybe you can use TreeMap for source hive table columns.